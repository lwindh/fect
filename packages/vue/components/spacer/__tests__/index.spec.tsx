import { mount } from '@vue/test-utils'

import Spacer from '../index'

describe('Spacer', () => {
  it('shoul be render as element', () => {
    const spacer = mount(Spacer)
    expect(spacer.html()).toMatchSnapshot()
  })

  it('should be support inline style', () => {
    const wrapper = mount(Spacer, {
      props: {
        inline: true,
      },
    })
    expect(wrapper.find('.fect-spacer').attributes('style')).toEqual('display: inline-block;')
  })

  it('should be support x and y', () => {
    const wrapper = mount({
      render() {
        return (
          <>
            <Spacer x={1} />
            <Spacer y={2} />
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be display right spacing when x or y are minus', () => {
    const wrapper = mount(Spacer, {
      props: {
        x: -0.5,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be display right spacing when x or y are string', () => {
    const wrapper = mount(Spacer, {
      props: {
        x: '111' as unknown as number,
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
