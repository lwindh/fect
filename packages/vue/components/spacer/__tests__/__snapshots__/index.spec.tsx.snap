// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer shoul be render as element 1`] = `<span class="fect-spacer" style="display: block;"></span>`;

exports[`Spacer should be display right spacing when x or y are minus 1`] = `<span class="fect-spacer" style="display: block;"></span>`;

exports[`Spacer should be display right spacing when x or y are string 1`] = `<span class="fect-spacer" style="display: block;"></span>`;

exports[`Spacer should be support x and y 1`] = `<span class="fect-spacer" style="display: block;"></span><span class="fect-spacer" style="display: block;"></span>`;
